### ISSUE TITLE
android: only disallow screenshots when seed is shown

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#8351]
Currently in the qml Android app, it is not possible to take screenshots.
This is done by setting [`FLAG_SECURE`](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SECURE) on the window. (see https://github.com/spesmilo/electrum/issues/8233, [p4a commit](https://github.com/accumulator/python-for-android/commit/b6e43aeb69bb7687a91703b7b8cd9af444e7f9f2))

I think this is overly restrictive. We only really want to prohibit taking a screenshot with the seed visible (or maybe private keys in general). Taking screenshots is useful for debugging and illustrating bugs and for explaining things to others.

I believe we should dynamically set/unset `FLAG_SECURE` when the seed is shown, instead of always having it enabled. I think this should be possible (although I have not tested yet).

**Comment by engineer22** [comment]
Today I had to find this restriction the hard way, had to find an app and download to create the QR code so I can scan it on the spot at the ATM I went.

Please make screenshots possible again.

**Comment by SomberNight** [comment]
I don't understand your use case. What exactly did you need another app for? Are you trying to buy or sell bitcoin at the ATM? Why would you need to take a screenshot in either case?

**Comment by engineer22** [comment]
@SomberNight sorry I mixed things up when writing.

Someone else was buying for me, and since I used to always send a screenshot of the QR code, I went to do the same only for the app not allowing me to. The other person had some problems typing the address manually so I had to find an app and pass the address to generate the QR code on my own.

**Comment by SomberNight** [comment]
I see, thanks for explaining.

**Comment by accumulator** [comment]
> Currently in the qml Android app, it is not possible to take screenshots. This is done by setting [`FLAG_SECURE`](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SECURE) on the window. (see #8233, [p4a commit](https://github.com/accumulator/python-for-android/commit/b6e43aeb69bb7687a91703b7b8cd9af444e7f9f2))
> 
> I think this is overly restrictive. We only really want to prohibit taking a screenshot with the seed visible (or maybe private keys in general). Taking screenshots is useful for debugging and illustrating bugs and for explaining things to others.
> 
> I believe we should dynamically set/unset `FLAG_SECURE` when the seed is shown, instead of always having it enabled. I think this should be possible (although I have not tested yet).

I've added a dynamic `FLAG_SECURE` toggle in https://github.com/accumulator/python-for-android/commit/87249fc06adaf75d8b6be8f732a21ac7e8774857
Some testing from the app suggests you can set the flag at any time, but clearing the flag appear to be ineffective.

**Comment by SomberNight** [comment]
Have you seen the hack in https://stackoverflow.com/a/43545487, to force a refresh?

```java
Window window = getActivity().getWindow();
WindowManager wm = getActivity().getWindowManager();

// Re-apply (re-draw) Window's DecorView so the change to the Window flags will be in place immediately.
if (flagsChanged && ViewCompat.isAttachedToWindow(window.getDecorView())) {
    // FIXME Removing the View and attaching it back makes visible re-draw on Android 4.x, 5+ is good.
    wm.removeViewImmediate(window.getDecorView());
    wm.addView(window.getDecorView(), window.getAttributes());
}
```

**Comment by accumulator** [comment]
ah cool, this looks usable!

**Comment by accumulator** [comment]
fixed in 0672ea20abd833feabb7fd0efa22ced99393ee1a
